packages/hawtio/src/plugins/shared/attributes/AttributeTable.tsx

```diff
@@ -0,0 +1,88 @@
+import { useContext, useEffect, useState } from 'react'
+import { Card, CardBody, Text } from '@patternfly/react-core'
+import { Table, TableBody, TableHeader, TableProps } from '@patternfly/react-table'
+import { PluginNodeSelectionContext } from '@hawtiosrc/plugins/selectionNodeContext'
+import { AttributeValues } from '@hawtiosrc/plugins/connect/jolokia-service'
+import { attributeService } from './attribute-service'
+import './AttributeTable.css'
+import { JmxContentMBeans } from '@hawtiosrc/plugins/shared/JmxContentMBeans'
+import { humanizeLabels } from '@hawtiosrc/util/strings'
+
+export const AttributeTable: React.FunctionComponent = () => {
+  const { selectedNode } = useContext(PluginNodeSelectionContext)
+  const [attributes, setAttributes] = useState<AttributeValues[]>([{}])
+  const [isReading, setIsReading] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (!selectedNode) {
+      return
+    }
+
+    const childrenMbeansAttributes: AttributeValues[] = []
+    const readAttributes = async () => {
+      setIsReading(true)
+      if (selectedNode.children)
+        for (const mbean of selectedNode.children) {
+          const objectName = mbean.objectName
+          if (objectName) {
+            const attrs = await attributeService.read(objectName)
+            childrenMbeansAttributes.push(attrs)
+          }
+        }
+      setAttributes([...childrenMbeansAttributes])
+      setIsReading(false)
+    }
+    readAttributes()
+  }, [selectedNode])
+
+  if (!selectedNode) {
+    return null
+  }
+
+  if (isReading) {
+    return (
+      <Card>
+        <CardBody>
+          <Text component='p'>Reading attributes...</Text>
+        </CardBody>
+      </Card>
+    )
+  }
+
+  const checkIfAllMBeansHaveSameAttributes = (attributes: AttributeValues[]) => {
+    if (attributes.length <= 1) {
+      return true
+    }
+
+    const firstAttributeElements = attributes[0].length
+
+    if (!attributes.every(attribute => attribute.length === firstAttributeElements)) {
+      return false
+    }
+
+    return attributes.every(attribute =>
+      Object.keys(attribute).every(label => Object.keys(attributes[0]).includes(label)),
```

Comment:
while performance shouldn't be of much problem, you could do a bit better here. It's O(N^2) right now.

1. You can use `Set` and `Set.has` for the first attributes
2. Put the first attributes set outside of the every loop

Then it could become O(N) per attributes. If you can find find a better way to compare the two sets in TS, it could be more efficient.